src/components/label/Label.tsx

```diff
@@ -1,25 +1,34 @@
-import React, { HTMLProps } from 'react';
+import React, { HTMLProps, useContext } from 'react';
 import classNames from 'classnames';
 import { NHSUKSize } from '../../util/types/NHSUKTypes';
+import FormGroupContext from '../formgroup/FormGroupContext';
 
 export interface LabelProps extends Omit<HTMLProps<HTMLLabelElement>, 'size'> {
   bold?: boolean;
   isPageHeading?: boolean;
   size?: NHSUKSize;
 }
 
-const BaseLabel: React.FC<LabelProps> = ({ className, bold, size, isPageHeading, ...rest }) => (
-  <label
-    className={classNames(
-      'nhsuk-label',
-      { 'nhsuk-label--s': bold && !size },
-      { 'nhsuk-label--xl': isPageHeading && !size },
-      { [`nhsuk-label--${size}`]: size },
-      className,
-    )}
-    {...rest}
-  />
-);
+const BaseLabel: React.FC<LabelProps> = props => {
+  const { id, className, bold, size, isPageHeading, ...rest } = props;
+  const { inputID } = useContext(FormGroupContext);
+  const labelId = props.id || inputID || undefined;
```

Comment:
The label ID wouldn't be just the `inputID` here - it would be `${inputID}--label`

Reply:
added ${inputID}--label